math/arctangent_test.go

```diff
@@ -1,170 +1,146 @@
+	} {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			z := decimal.WithPrecision(N)
+			x, _ := new(decimal.Big).SetString(tt.x)
+			r, _ := new(decimal.Big).SetString(tt.r)
+
+			Atan(z, x)
+			if z.Cmp(r) != 0 && diff.Sub(r, z).CmpAbs(eps) > 0 {
+				t.Errorf(`#%d: Atan(%s)
+wanted: %s
+got   : %s
+diff  : %s
+`, i, x, r, z, diff)
 			}
 		})
 	}
 }
 
+var atan_X, _ = new(decimal.Big).SetString("0.7853981633974483096156608458198757210492923498437764552437361480769541015715522496570087063355292670")
+
 func BenchmarkAtan(b *testing.B) {
-	precision := 30
-	four := decimal.New(4, 0)
-	piOver4 := Pi(decimal.WithPrecision(precision))
-	piOver4.Quo(piOver4, four)
-	result := decimal.WithPrecision(precision)
-	b.ResetTimer()
-
-	for i := 0; i < b.N; i++ {
-		Atan(result, piOver4)
+	for _, prec := range benchPrecs {
+		b.Run(fmt.Sprintf("%d", prec), func(b *testing.B) {
+			z := decimal.WithPrecision(prec)
+			for j := 0; j < b.N; j++ {
+				Atan(z, atan_X)
+			}
+			gB = z
+		})
 	}
 }
 
 func TestAtan2(t *testing.T) {
-	naN := new(decimal.Big).SetNaN(false)
-	naN.Context.Conditions |= decimal.InvalidOperation
-	zeroNeg, _ := new(decimal.Big).SetString("-0")
-	piPos := Pi(new(decimal.Big))
-	piNeg := new(decimal.Big).Neg(piPos)
-	oneNeg := new(decimal.Big).Neg(one)
-	piOver2 := Pi(new(decimal.Big))
-	piOver2.Quo(piOver2, two)
-	piOver2Neg := new(decimal.Big).Neg(piOver2)
-	infPlus := new(decimal.Big).SetInf(false)
-	infNeg := new(decimal.Big).SetInf(true)
-	piOver4 := Pi(new(decimal.Big))
-	piOver4.Quo(piOver4, four)
-	piOver4Neg := new(decimal.Big).Neg(piOver4)
-	threePiOver4 := new(decimal.Big).Mul(piOver4, three)
-	threePiOver4Neg := new(decimal.Big).Mul(piOver4Neg, three)
-	oneOverOneAtan := Atan(new(decimal.Big), one)
-	oneOverOneNegAtan := Atan(new(decimal.Big), new(decimal.Big).Quo(one, oneNeg))
-	zeroOverOneNegAtan := Atan(new(decimal.Big), new(decimal.Big).Quo(zero, oneNeg))
-	oneNegOverOneAtan := Atan(new(decimal.Big), new(decimal.Big).Quo(oneNeg, one))
-	oneNegOverOneNegAtan := Atan(new(decimal.Big), new(decimal.Big).Quo(oneNeg, oneNeg))
-	zeroOverOneNegAtanPlusPi := new(decimal.Big).Add(zeroOverOneNegAtan, Pi(new(decimal.Big)))
-	oneOverOneNegAtanPlusPi := new(decimal.Big).Add(oneOverOneNegAtan, Pi(new(decimal.Big)))
-	oneNegOverOneNegAtanMinusPi := new(decimal.Big).Sub(oneNegOverOneNegAtan, Pi(new(decimal.Big)))
-	type args struct {
-		z *decimal.Big
-		y *decimal.Big
-		x *decimal.Big
-	}
-	tests := []struct {
-		name string
-		args args
-		want *decimal.Big
+	const N = 1000
```

Comment:
This is over kill.  As most of the this test is logic based the standard 16 is sufficient.  Atan is tested above.